### ISSUE TITLE
How to determine whether the loss of the yolo model is at the local optimal solution during training and whether it is at the saddle point

### ISSUE THREAD

**Issue by hhhFuture** [ultralytics/ultralytics#20981]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

I plan to use yolov8n to train an object detection model that can recognize scissors gestures. To reduce the misrecognition of fists, palms and ok gestures, I added about 3 percent of the background image. During the training, box_loss and cls_loss kept oscillating back and forth at a certain value, so I was wondering whether it was at the local optimal solution and whether it was at the saddle point. I checked the official documentation of ultralytics. The momentum parameter is the momentum factor for SGD or beta1 for the Adam optimizer, which is used to incorporate past gradients into the current update. However, during training, the rotation of the optimizer is set to auto. I have no idea which optimizer it uses or whether it uses momentum. I opened the args.yaml generated during training, hoping to learn about the specific parameter information during training through this file, but the optimizer inside also showed "auto". So I really want to know whether the model was already at the optimal solution or at the saddle point when I was training. The following are the parameters I set during training. If possible, I sincerely hope to receive help from all my friends. I would like to thank all of you in advance

(Because I trained based on the last time of the previous model, the learning rate was set to 0.005. There are already samples of small targets in my dataset, so the scale is set to 0.2. I'm also wondering if it's because the learning rate is set too small)

Training configuration：
from ultralytics import YOLO
import time

if __name__ == '__main__':
    model = YOLO(r"E:\Project_Gesture\model_script\runs\detect\scissors_yolo8n3\weights\best.pt")
    model.train(
        data=r"E:\Project_Gesture\model_script\scissors.yaml",
        imgsz=640,
        device=0,
        lr0=0.005, 
        epochs=200,
        batch=64,
        close_mosaic=30,
        name="scissors_yolo8n",
        fliplr=0.5,  
        flipud=0.5,  
        degrees=15, 
        mosaic=0.5, 
        mixup=0.3,  
        plots=True,  
        scale=0.2, 
    )

Screenshots of the training：

![Image](https://github.com/user-attachments/assets/2667d82d-519d-40ca-908a-c5bd4a6d7cc1)

![Image](https://github.com/user-attachments/assets/10a8e921-41f4-4441-bc18-7ef06a5510e7)

![Image](https://github.com/user-attachments/assets/71140b71-2271-4dc1-a70b-79740e646c4e)

**Comment by UltralyticsAssistant** [comment]
👋 Hello @hhhFuture, thank you for your detailed question and for sharing your training configuration and screenshots! This is an automated response to help you get started—an Ultralytics engineer will review and assist you soon.

For questions regarding training dynamics, optimizer settings, or loss behavior, we recommend checking the [Docs](https://docs.ultralytics.com/) for additional insights into [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) workflows, as well as the [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/). You can also find guidance on optimizer selection and parameter details there.

To help us provide the most accurate support, please consider:
- Including a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) (MRE) if you notice unexpected behavior or suspect a bug during training 🐛
- Sharing any sample images or additional logs if you have specific concerns about training performance or results

For community support and live discussions, join us on [Discord](https://discord.com/invite/ultralytics) 🎧, or participate on [Discourse](https://community.ultralytics.com/) and our [Subreddit](https://reddit.com/r/Ultralytics).

## Upgrade

Make sure you are running the latest `ultralytics` package with all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) installed in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

You can run YOLO models in any of the following up-to-date, verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/), and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Thank you for contributing to Ultralytics! Your patience is appreciated while we review your question.

**Comment by Y-T-G** [comment]
It shows the optimizer in the training log before the training starts
